app/lib/classification-queue.js

```diff
@@ -40,43 +43,35 @@ class ClassificationQueue {
   }
 
   flushToBackend() {
-    const queue = this._loadQueue();
-
-    if (queue.length > 0) {
-      console.log('Saving queued classifications:', queue.length);
-      queue.forEach((classificationData) => {
-        this.apiClient.type('classifications').create(classificationData).save().then((actualClassification) => {
+    const pendingClassifications = this._loadQueue();
+    const failedClassifications = [];
+    this._saveQueue(failedClassifications);
+
+    if (pendingClassifications.length > 0) {
+      if (process.env.BABEL_ENV !== 'test') console.log('Saving queued classifications:', pendingClassifications.length);
+      return Promise.all(pendingClassifications.map((classificationData) => {
+        return this.apiClient.type('classifications').create(classificationData).save()
```

Comment:
I think I'm seeing errors here - apparently `this.apiClient` doesn't exist? Debugging in process.

Reply:
Wait, sorry, it's `this.apiClient.type` that doesn't exist.

It seems that `this.apiClient` is the following object:

```
Storage {failed-classifications: "[]", minilogSettings: "true", pusherTransportEncrypted: "{"timestamp":1538059649538,"transport":"ws","latency":628}", length: 3}
```

Huuhhh?

Reply:
Dammit. Why is that not erroring when running the tests. `npm run test-one app/lib/classification-queue.spec.js`

Reply:
It's the `constructor`

This PR's updated `constructor` call changes it from `constructor(storage, api, onClassificationSaved)` to `constructor(api, onClassificationSaved)`, but the defining call of `new QueueStorage` (see `pages/projects/classify.jsx`) still defines some sort of Storage object as the first argument.

Reply:
D'oh!

Reply:
By the way, I'm going to rebase this because a bunch of packages have been updated recently.

Reply:
That extra storage argument should be gone now.